File: DayOne/Java/Level1/PerfectSquare.java
import java.util.Arrays;
public class PerfectSquare{
    public static boolean isPerfectSquare(int number){
        boolean check = false;
        if(number < 0){
            check = false;
        }
        int squareRoot = (int)Math.sqrt(number);
        if(squareRoot * squareRoot == number){
            check = true;
        }
        return check;
    }
    
    public static int[] getPerfectSquare(int[] array){
        int countPerfectSquare = 0;
        int count = 0;
        for(int index = 0; index < array.length; index++){
            if(isPerfectSquare(array[index])){
                countPerfectSquare++;
            }
        }
        int[] perfectSquareArray = new int[countPerfectSquare];
        for(int index = 0; index < array.length; index++){
            if(isPerfectSquare(array[index])){
                perfectSquareArray[count] = array[index];
                count++;    
            }
        }
        return perfectSquareArray;
    }
    
    
     
    public static void main(String[] args){
        int[] array = {4, 0, 9, 0, 16, 18};
        System.out.println(Arrays.toString(getPerfectSquare(array)));
                
    }
    



}

File: DayOne/Python/Level1/perfect_square.py
import math

def is_perfect_square(number):
    check = False
    if number < 0:
        check = False
        return check 
    square_root = int(math.sqrt(number))
    if square_root * square_root == number:
        check = True
        
    return check

def get_perfect_square(array):
    perfect_square_array = []
    for index in range(len(array)):
        if is_perfect_square(array[index]):
            perfect_square_array.append(array[index])
            
    return perfect_square_array


